jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ImageParameters.java

```diff
@@ -39,8 +39,12 @@
   @Nullable private String credHelper;
 
   @Inject
-  public ImageParameters(ObjectFactory objectFactory) {
-    auth = objectFactory.newInstance(AuthParameters.class);
+  public ImageParameters(ObjectFactory objectFactory, String imageDescriptor) {
+    auth =
+        objectFactory.newInstance(
+            AuthParameters.class,
+            imageDescriptor + ".auth.username",
```

Comment:
Oh, right this can actually be just `descriptor + ".auth"`, and then `AuthParameters` adds the `username` and `password` suffixes. Essentially, the idea is to pass along the descriptor name for each of the parameters at the level at which it is applicable (and the same for Maven).